src/frontends/tensorflow_common/src/op/angle.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "common_op_table.hpp"
+#include "helper_ops/complex_type_mark.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/atan.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert_like.hpp"
+#include "openvino/op/divide.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/greater.hpp"
+#include "openvino/op/greater_eq.hpp"
+#include "openvino/op/less.hpp"
+#include "openvino/op/logical_and.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/select.hpp"
+#include "openvino/op/subtract.hpp"
+
+using namespace std;
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+
+OutputVector translate_angle_op(const NodeContext& node) {
+    default_op_checks(node, 1, {"Angle"}, true);
+    auto complex = node.get_input(0);
+
+    auto complex_type_mark = as_type_ptr<ComplexTypeMark>(complex.get_node_shared_ptr());
+    element::Type complex_part_type = complex_type_mark->get_complex_part_type();
+    auto op_type = node.get_op_type()
+    auto coomplex = complex_type_mark->input_value(0);
+
+    TENSORFLOW_OP_VALIDATION(
+    node,
+    complex_type_mark,
+    "[TENSORFLOW FRONTEND] Internal error: complex_type_mark does not support" + op_type + "operation"
+    );
+
+    auto real_index = make_shared<v0::Constant>(element::i32, Shape{1}, 0);
+    auto imag_index = make_shared<v0::Constant>(element::i32, Shape{1}, 1);
```

Comment:
```suggestion
    auto real_index = make_shared<v0::Constant>(element::i32, Shape{}, 0);
    auto imag_index = make_shared<v0::Constant>(element::i32, Shape{}, 1);
```
we should not preserve auxiliary last dimension. It should be eliminated during `Gather` operation execution